pyanaconda/ui/lib/payload.py

```diff
@@ -56,3 +57,58 @@ def get_payload(payload_type=PAYLOAD_TYPE_DNF):
 
     # Or create a new payload.
     return create_payload(payload_type)
+
+
+def create_source(source_type):
+    """Create a source.
+
+    :param source_type: a source type
+    :return: a DBus proxy of a source
+    """
+    payloads_proxy = PAYLOADS.get_proxy()
+    object_path = payloads_proxy.CreateSource(source_type)
+    return PAYLOADS.get_proxy(object_path)
+
+
+def set_source(payload_proxy, source_proxy):
+    """Attach the source to the payload.
+
+    :param payload_proxy: a DBus proxy of a payload
+    :param source_proxy: a DBus proxy of a source
+    """
+    object_path = get_object_path(source_proxy)
+    payload_proxy.SetSources([object_path])
+
+
+def get_source(payload_proxy, source_type=None):
```

Comment:
Please rename `source_type` to `default_source_type`. This way it seems to return source of the given type.

Reply:
Fixed.